tests/unit/test_oauth_token.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+from freezegun import freeze_time
+
+from neptune_api.errors import InvalidApiTokenException
+from neptune_api.types import OAuthToken
+
+
+def test_invalid():
+    # given
+    access_token = "eyJ"
+    refresh_token = ""
+
+    # then
+    with pytest.raises(InvalidApiTokenException):
+        OAuthToken.from_tokens(access=access_token, refresh=refresh_token)
+
+
+@freeze_time("2019-10-25 12:36:04")
+def test_expiration_time():
+    # given
+    access_token = "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6Ik5FRTFRVVJCT1RNNE16STVSa0ZETlRZeE9UVTFNRGcyT0Rnd1EwVXpNVGsxUWpZeVJrUkZRdyJ9.eyJpc3MiOiJodHRwczovL2Rldi04N2V2eDlydS5hdXRoMC5jb20vIiwic3ViIjoiYVc0Q2NhNzl4UmVMV1V6MGFFMkg2a0QwTzNjWEJWdENAY2xpZW50cyIsImF1ZCI6Imh0dHBzOi8vZXhwZW5zZXMtYXBpIiwiaWF0IjoxNTcyMDA2OTU0LCJleHAiOjE1NzIwMDY5NjQsImF6cCI6ImFXNENjYTc5eFJlTFdVejBhRTJINmtEME8zY1hCVnRDIiwiZ3R5IjoiY2xpZW50LWNyZWRlbnRpYWxzIn0.PUxE7xn52aTCohGiWoSdMBZGiYAHwE5FYie0Y1qUT68IHSTXwXVd6hn02HTah6epvHHVKA2FqcFZ4GGv5VTHEvYpeggiiZMgbxFrmTEY0csL6VNkX1eaJGcuehwQCRBKRLL3zKmA5IKGy5GeUnIbpPHLHDxr-GXvgFzsdsyWlVQvPX2xjeaQ217r2PtxDeqjlf66UYl6oY6AqNS8DH3iryCvIfCcybRZkc_hdy-6ZMoKT6Piijvk_aXdm7-QQqKJFHLuEqrVSOuBqqiNfVrG27QzAPuPOxvfXTVLXL2jek5meH6n-VWgrBdoMFH93QEszEDowDAEhQPHVs0xj7SIzA"
+    refresh_token = ""
+
+    # when
+    token = OAuthToken.from_tokens(access=access_token, refresh=refresh_token)
+
+    # then
+    assert token.seconds_left == 0.0
+    assert token.is_expired is True
+
+
+@freeze_time("2019-10-25 12:35:34")
+def test_expired():
+    # given
+    access_token = "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsImtpZCI6Ik5FRTFRVVJCT1RNNE16STVSa0ZETlRZeE9UVTFNRGcyT0Rnd1EwVXpNVGsxUWpZeVJrUkZRdyJ9.eyJpc3MiOiJodHRwczovL2Rldi04N2V2eDlydS5hdXRoMC5jb20vIiwic3ViIjoiYVc0Q2NhNzl4UmVMV1V6MGFFMkg2a0QwTzNjWEJWdENAY2xpZW50cyIsImF1ZCI6Imh0dHBzOi8vZXhwZW5zZXMtYXBpIiwiaWF0IjoxNTcyMDA2OTU0LCJleHAiOjE1NzIwMDY5NjQsImF6cCI6ImFXNENjYTc5eFJlTFdVejBhRTJINmtEME8zY1hCVnRDIiwiZ3R5IjoiY2xpZW50LWNyZWRlbnRpYWxzIn0.PUxE7xn52aTCohGiWoSdMBZGiYAHwE5FYie0Y1qUT68IHSTXwXVd6hn02HTah6epvHHVKA2FqcFZ4GGv5VTHEvYpeggiiZMgbxFrmTEY0csL6VNkX1eaJGcuehwQCRBKRLL3zKmA5IKGy5GeUnIbpPHLHDxr-GXvgFzsdsyWlVQvPX2xjeaQ217r2PtxDeqjlf66UYl6oY6AqNS8DH3iryCvIfCcybRZkc_hdy-6ZMoKT6Piijvk_aXdm7-QQqKJFHLuEqrVSOuBqqiNfVrG27QzAPuPOxvfXTVLXL2jek5meH6n-VWgrBdoMFH93QEszEDowDAEhQPHVs0xj7SIzA"
+    refresh_token = ""
+
+    # when
+    token = OAuthToken.from_tokens(access=access_token, refresh=refresh_token)
+
+    # then
+    assert token.seconds_left == 30.0
```

Comment:
what is the meaning of `seconds_left` because imo it implies that its the ammount of seconds for which token will be still valid

Reply:
Yes, maybe `valid_for_seconds` would be better?

Reply:
yeah `valid_for_seconds` could be a better choice or we could change

```
  @property
    def seconds_left(self) -> float:
        return self._expiration_time - time.time()
```
to
```
  @property
    def seconds_left(self) -> float:
        return self._expiration_time - time.time() - MINIMAL_EXPIRATION_SECONDS
```

and then `seconds_left` would be accurate so

```
    @property
    def is_expired(self) -> bool:
        return self.seconds_left <= MINIMAL_EXPIRATION_SECONDS
```
could be change to
```
    @property
    def is_expired(self) -> bool:
        return self.seconds_left <= 0
 ```

Reply:
Sure, done